modules/gitops-argocd-agent-install-principal-component.adoc

```diff
@@ -0,0 +1,314 @@
++
+[source,yaml]
+----
+spec:
+  sourceNamespaces:
+  - "agent-managed"
+  - "agent-autonomous"
+----
++
+If you modify the `AppProject`, restart the Argo CD pods to apply the changes.
+
+. Create Required Secrets.
+The principal component uses a Public Key Infrastructure (PKI) to secure all communications.
++
+[NOTE]
+====
+The CLI-generated PKI is intended for development and testing purposes only. For production environments, use certificates issued by your organization's PKI or a trusted certificate authority.
+====
+
+.. Initialize the PKI
+
+... To initialize the Certificate Authority (CA) that signs other certificates, run the following command:
++
+[source,terminal]
+----
+$ argocd-agentctl pki init \
+     --principal-namespace "<PRINCIPAL_NAMESPACE>" \
+     --principal-context "<PRINCIPAL_CONTEXT>"
+----
++
+where:
+
+`<f>`:: Overwrites the existing CA if one already exists.
+`<principal-namespace>`:: Specifies the namespace where the Principal component is installed.
+`<principal-context>`:: Specifies the context of the Principal component.
++
+This command generates a self-signed CA certificate and private key, and creates the `argocd-agent-ca` secret that contains the CA credentials. The CA is valid for signing certificates and includes a default expiration period.
+
+... To generate the server certificate for the Principal's gRPC service, run the following command:
++
+[source,terminal]
+----
+$ argocd-agentctl pki issue principal
+--principal-namespace "<PRINCIPAL_NAMESPACE>"
+--principal-context "<PRINCIPAL_CONTEXT>"
+--dns "<PRINCIPAL_DNS_NAME>"
+----
++
+where:
+
+`<ip>`:: Specifies the comma-separated list of IP addresses where the Principal component is accessible.
+`<dns>`:: Specifies the comma-separated list of DNS names for the Principal service.
+`<u>`, `<upsert>`:: Updates the existing certificate if one already exists.
+`<principal-namespace>`:: Specifies the namespace where the Principal component is installed.
+`<principal-context>`:: Specifies the context of the Principal component.
++
+This command issues a TLS certificate for the Principal to enable secure gRPC communication between the Principal and Agent components. It also creates the `argocd-agent-principal-tls` secret that contains the generated certificate and key.
+
+... The resource proxy enables Argo CD to access live resources on remote clusters through the Principal component. It requires its own certificate for mutual TLS (mTLS) authentication. To generate the server certificate for the resource proxy service, run the following command:
++
+[source,terminal]
+----
+$ argocd-agentctl pki issue resource-proxy
+--principal-namespace "<PRINCIPAL_NAMESPACE>"
+--principal-context "<PRINCIPAL_CONTEXT>"
+--dns "<RESOURCE_PROXY_DNS>"
+----
++
+where:
+
+`<ip>`:: Specifies the comma-separated list of IP addresses for the resource proxy that are reachable by the Argo CD API server.
+`<dns>`:: Specifies the DNS name for the Principal service.
```

Comment:
From Jayendra's doc, additional context for what value to place in dns:
> --dns: Service name for resource proxy in hub cluster (ex: openshift-gitops-agent-principal-resource-proxy). Argo CD uses this to connect with resource-proxy. Since both Argo CD and the resource-proxy are in the same cluster, Argo CD can connect using the service name itself.

Reply:
+1 Jonathan, 
Dhruv please replace "openshift-gitops-agent-principal-resource-proxy" with "(ArgoCD instance name)-agent-principal-resource-proxy" in description given in my doc.